src/global.rs

```diff
@@ -1,46 +1,51 @@
+use crate::AsyncRayonHandle;
+use std::panic::{catch_unwind, AssertUnwindSafe};
 use tokio::sync::oneshot;
-use tokio::sync::oneshot::error::RecvError;
 
 /// Asynchronous wrapper around Rayon's [`spawn`](rayon::spawn).
 ///
 /// Runs a function on the global Rayon thread pool with LIFO priority,
 /// produciing a future that resolves with the function's return value.
 ///
-/// # Errors
-/// Forwards Tokio's [`RecvError`](tokio::sync::oneshot::error::RecvError), i.e. if the channel is closed.
-pub async fn spawn_async<F, R>(func: F) -> Result<R, RecvError>
+/// # Panics
+/// If the task function panics, the panic will be propagated through the
+/// returned future. Thie will NOT trigger the Rayon thread pool's panic
+/// handler.
```

Comment:
Rayon's panic handler can actually be triggered: if the receivers is dropped before it finishes, the return value is dropped in the rayon thread, and that destructor may panic.

Reply:
Oh yeah - if you dropped the `AsyncHandle`, that would close the channel, and the return value of `func` would be dropped.

That feels like enough of an edge case that we could probably just document it, right? I can't think of any code solutions other than swallowing the destructor panic, which is obviously not great.

Reply:
Added a blurb about that scenario to the docs!